### PR TITLE
Support for vanilla animation tables

### PR DESCRIPTION
**Pull request by Lilaa3**
Added support for vanilla animation tables.

### CODE REVIEW

File: fast64_internal/sm64/sm64_anim.py [hunk 1]
```diff
@@ -176,12 +176,41 @@ def exportAnimationC(armatureObj, loopAnim, dirPath, dirName, groupName,
...
+		if footerIndex == -1:
+			footerIndex = stringData.rfind('};')
+
+			# if that can´t be found then throw an error.
+			if footerIndex == -1:
+				raise PluginError("Animation table´s footer does not seem to exist.")
+			
+			stringData = stringData[:footerIndex] + '\tNULL,\n' + stringData[footerIndex:]
+
+		stringData = stringData[:footerIndex] + '\t&{0},\n'.format(sm64_anim.header.name) + stringData[footerIndex:]
```
Comment (thecozies) [inline fast64_internal/sm64/sm64_anim.py:209]: we've mostly moved to using f-strings when possible

```suggestion
		stringData = stringData[:footerIndex] + f'\t&{sm64_anim.header.name},\n' + stringData[footerIndex:]
```

**Review by thecozies — CHANGES_REQUESTED** [review]
smol suggestion

File: fast64_internal/sm64/sm64_anim.py [hunk 2]
```diff
@@ -176,12 +176,41 @@ def exportAnimationC(armatureObj, loopAnim, dirPath, dirName, groupName,
...
 		tableFile = open(tableFilePath, 'w', newline='\n')
 		tableFile.write('const struct Animation *const ' + \
 			animsName + '[] = {\n\tNULL,\n};\n')
 		tableFile.close()
-	writeIfNotFound(tableFilePath, '\t&' + sm64_anim.header.name + ',\n', '\tNULL,\n};')
+		
+	fileData = open(tableFilePath, 'r')
+	fileData.seek(0)
+	stringData = fileData.read()
+	fileData.close()
```
Comment (Dragorn421) [inline fast64_internal/sm64/sm64_anim.py:190]:
```suggestion
	with open(tableFilePath, 'r') as f:
		stringData = f.read()
```
(if you want to read, see
https://docs.python.org/3/tutorial/inputoutput.html#tut-files
https://docs.python.org/3/reference/datamodel.html#context-managers )

File: fast64_internal/sm64/sm64_anim.py [hunk 3]
```diff
@@ -176,12 +176,41 @@ def exportAnimationC(armatureObj, loopAnim, dirPath, dirName, groupName,
...
+			if footerIndex == -1:
+				raise PluginError("Animation table´s footer does not seem to exist.")
+			
+			stringData = stringData[:footerIndex] + '\tNULL,\n' + stringData[footerIndex:]
+
+		stringData = stringData[:footerIndex] + f'\t&{sm64_anim.header.name},\n' + stringData[footerIndex:]
+		
+		fileData = open(tableFilePath, 'w', newline = '\n')
+		fileData.write(stringData)
+		fileData.close()
```
Comment (Dragorn421) [inline fast64_internal/sm64/sm64_anim.py:213]:
```suggestion
		with open(tableFilePath, 'w', newline = '\n') as f:
			f.write(stringData)
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
I'm not a sm64 person so I can't tell as for what this achieves, just two comments on the python. Sorry for the delay

(and @thecozies bump?)

File: fast64_internal/sm64/sm64_anim.py (hunk 2 above)
Comment (Dragorn421) [reply fast64_internal/sm64/sm64_anim.py]: you're a lot less late than we are on the review :sweat_smile: , and to me not late at all anyway, few days of delay is perfectly fine

**Review by Dragorn421 — APPROVED** [review]
code is okay to me, but I'm not a sm64 person

**Review by thecozies — APPROVED** [review]
# nice

**Pull request merged**